packages/plugin-data/src/commands/force/data/soql/query.ts

```diff
@@ -0,0 +1,140 @@
+   *
+   * @protected
+   */
+  protected static readonly result: SfdxResult = {
+    display(): void {
+      const results = asPlainObject(this.data) as DataSoqlQueryResult;
+      let reporter;
+      switch (results.resultFormat as keyof typeof FormatTypes) {
+        case 'human':
+          reporter = new HumanReporter(results, results.columns, this.ux, results.logger);
+          break;
+        case 'json':
+          reporter = new JsonReporter(results, results.columns, this.ux, results.logger);
+          break;
+        case 'csv':
+          reporter = new CsvReporter(results, results.columns, this.ux, results.logger);
+          break;
+        default:
+          throw new Error(`result format is invalid: ${results.resultFormat}`);
+      }
+      // delegate to selected reporter
+      reporter.display();
+    },
+  };
+
+  // Overrides SfdxCommand.  This is ensured since requiresUsername == true
+  protected org!: Org;
+
+  /**
+   * Command run implementation
+   *
+   * Returns either a DataSoqlQueryResult or a SfdxResult.
+   * When the user is using global '--json' flag an instance of SfdxResult si returned.
+   * This is necessary since '--json' flag reports results in the form of SfdxResult
+   * and bypasses the definition of start result. The goal is to have the output
+   * from '--json' and '--resulformat json' be the same.
+   *
+   * The DataSoqlQueryResult is necessary to communicate user selections to the reporters.
+   * The 'this' object available during display() function does not include user input to
+   * the command, which are necessary for reporter selection.
+   *
+   */
+  public async run(): Promise<DataSoqlQueryResult | SfdxResult> {
+    try {
+      if (this.flags.resultformat !== 'json') this.ux.startSpinner(messages.getMessage('queryRunningMessage'));
+      const query = new SoqlQuery(
+        this.getConnection(this.org, this.flags.usetoolingapi),
+        this.flags.query,
+        this.logger
+      );
+      const queryResult: SoqlQueryResult = await query.runSoqlQuery();
+      const results = {
+        ...queryResult,
+        resultFormat: this.flags.resultformat,
+        json: this.flags.json,
+        logger: this.logger,
+      };
+      return this.normalizeIfJson(results);
+    } finally {
+      if (this.flags.resultformat !== 'json') this.ux.stopSpinner();
+    }
+  }
+
+  /**
+   * This function maps the DataSoqlQueryResult instance to the the well known 'data' property in
+   * SfdxResult class when the user has selected '--json' flag.
+   *
+   * @param results
+   * @private
+   */
+  private normalizeIfJson(results: DataSoqlQueryResult): DataSoqlQueryResult | SfdxResult {
+    if (this.flags.json) {
+      return {
+        data: toJsonMap(results.result),
+      };
+    }
+    return results;
+  }
+
+  private getConnection(org: Org, useToolingApi: boolean): Connection | Tooling {
+    return useToolingApi ? org.getConnection().tooling : org.getConnection();
+  }
```

Comment:
Is this a method for easier testing?  With it being just 1 line, seems like it could just go within `run()` and maybe you wouldn't need to import `Tooling` in that case?

Reply:
I will inline the ternary and remove the private function.